### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by l3r8yJ**
@h1alexbel take a look, please

wdyt?

---

### Focus of this PR:
This PR focuses on making various changes and improvements to the codebase.

### Detailed summary:
- Added `target` and `.idea` to `.gitignore`.
- Updated `AuthorTest.java` to use `asString()` instead of `value()`.
- Updated `XpathList.java` to remove `throws Exception` from `value()`.
- Updated `Sha.java` to remove `throws Exception` from `value()` and `close()`.
- Updated `Cql.java` to remove `throws Exception` from `apply()`.
- Updated `CassandraRunsIT.java` to use `CassandraIntegration.CASSANDRA` instead of `CASSANDRA`.
- Updated `Names.java` to remove `throws Exception` from `value()`.
- Updated `InFile.java` to remove `throws Exception` from `apply()` and convert `asString()` to `toString()`.
- Updated `InFileIT.java` to use `CassandraIntegration.HOST` instead of `CassandraIntegration.host`.
- Updated `SimpleIT.java` to use `CassandraIntegration.HOST` instead of `CassandraIntegration.host`.
- Updated `Simple.java` to remove `throws IOException` from `close()`.
- Added import for `io.github.eocqrs.cmig.sha.StateChanges` in `State.java`.
- Updated `Author.java` to implement `Text` instead of `Scalar<String>`.
- Updated `CassandraIntegration.java` to use `@AfterAll` instead of `@AfterClass` and update variable name to `HOST`.
- Updated `StateChanges.java` to remove unused imports, refactor `value()` method, and add `contentOf()` method.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by zoeself** [comment]
@l3r8yJ thank you for your Pull Request. I'll assign someone to review it soon. 

 If this PR solves a ``todo`` from the code, please don't forget to remove it.

**Comment by zoeself** [comment]
@h1alexbel please review this Pull Request. Deadline (when it should be merged or closed) is ``2023-08-17T07:49:05.846396``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

File: src/main/java/io/github/eocqrs/cmig/State.java [hunk 1]
```diff
@@ -22,19 +22,17 @@
...
 
 /**
  * State.
  *
  * @author Aliaksei Bialiauski (abialiauski.dev@gmail.com)
  * @since 0.0.0
  */
-public interface State extends Scalar<String> {
-
+public interface State extends Scalar<StateChanges> {
```
Comment (h1alexbel) [inline src/main/java/io/github/eocqrs/cmig/State.java:34]: @l3r8yJ can you provide simple implementation of this interface?

**Review by h1alexbel — COMMENTED** [review]
@l3r8yJ looks good

File: src/main/java/io/github/eocqrs/cmig/State.java (hunk 1 above)
Comment (h1alexbel) [reply src/main/java/io/github/eocqrs/cmig/State.java:34]: @l3r8yJ it will help to resolve this one: https://github.com/eo-cqrs/cmig/issues/36

Comment (h1alexbel) [reply src/main/java/io/github/eocqrs/cmig/State.java:34]: @l3r8yJ also I have a suggestion for the design of this interface: we can add:
```java
Author author();
```

That will return a particular ChangeState's author from XML file

WDYT?

Comment (l3r8yJ) [reply src/main/java/io/github/eocqrs/cmig/State.java:34]: @h1alexbel I think `Author author()` is an object with outside state, which I really don't like, wdyt?

I'm not sure about the implementation yet, I think I'll create a puzzle, wdyt?

Comment (h1alexbel) [reply src/main/java/io/github/eocqrs/cmig/State.java:34]: @l3r8yJ ChangeState has files a.k.a StateChanges and Author of it
ok, go ahead and create a puzzle

**Comment by h1alexbel** [comment]
@l3r8yJ let's integrate it in the master branch right after #42

**Comment by h1alexbel** [comment]
@l3r8yJ can you resolve the conflict and we are good to go

**Comment by l3r8yJ** [comment]
@h1alexbel sure, bit later

**Comment by zoeself** [comment]
@h1alexbel Don't forget to close this ticket before the deadline (2023-08-17T07:49:06). You are past the first half of the allowed period.

**Review by h1alexbel — APPROVED** [review]

**Comment by zoeself** [comment]
@h1alexbel thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**